neqo-transport/src/connection.rs

```diff
@@ -1275,25 +1297,53 @@ impl Connection {
             self.flow_mgr.borrow_mut().set_need_close_frame(false);
         }
 
-        // Sent a probe pkt. Another timeout will re-engage ProbeTimeout mode,
-        // but otherwise return to honoring CC.
-        if self.tx_mode == TxMode::Pto {
-            self.tx_mode = TxMode::Normal;
-        }
-
         if out_bytes.is_empty() {
-            assert!(self.tx_mode != TxMode::Pto);
             Ok(None)
         } else {
             // Pad Initial packets sent by the client to mtu bytes.
             if self.role == Role::Client && needs_padding {
                 qdebug!([self], "pad Initial to max_datagram_size");
                 out_bytes.resize(path.mtu(), 0);
             }
+            if let TxMode::Pto { dgram_avail, .. } = &mut self.tx_mode {
+                assert!(*dgram_avail > 0);
+                *dgram_avail -= 1;
```

Comment:
I'm not sure that I understand the use of dgram_avail.  Can you comment this with some explanations.

Reply:
I will refactor this and remove tx_mode from connection.